scapy/contrib/nsh.py

```diff
@@ -29,25 +29,19 @@
 
 #
 # NSH Support
-# https://www.ietf.org/id/draft-ietf-sfc-nsh-05.txt
+# https://www.rfc-editor.org/rfc/rfc8300.txt  January 2018
 #
 
 
-class Metadata(Packet):
-    name = 'NSH metadata'
-    fields_desc = [XIntField('value', 0)]
-
-
 class NSHTLV(Packet):
     "NSH MD-type 2 - Variable Length Context Headers"
     name = "NSHTLV"
     fields_desc = [
         ShortField('Class', 0),
-        BitField('Critical', 0, 1),
-        BitField('Type', 0, 7),
-        BitField('Reserved', 0, 3),
-        BitField('Len', 0, 5),
-        PacketListField('Metadata', None, XIntField, count_from='Len')
+        BitField('Type', 0, 8),
+        BitField('Reserved', 0, 1),
+        BitField('Len', 0, 7),
```

Comment:
While you are doing non backward compatibility changes: could you take the opportunity to switch the name of all fields that only have a capital letter at their beginning (not fully capitalized ones) to lowercase ?

That's the default in Scapy. Especially `len` is always writen as so